cli/kata-vmcache/main.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2019 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"path/filepath"
+	rDebug "runtime/debug"
+
+	google_protobuf "github.com/golang/protobuf/ptypes/empty"
+	"github.com/kata-containers/runtime/pkg/katautils"
+	pb "github.com/kata-containers/runtime/protocols/cache"
+	vc "github.com/kata-containers/runtime/virtcontainers"
+	vf "github.com/kata-containers/runtime/virtcontainers/factory"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc"
+)
+
+var usage = fmt.Sprintf(`%s kata VM cache server
+
+When enable_vm_cache enabled, %s start the VM cache server
+that created some VMs (number is set by option number) as
+VM cache.
+Each kata-runtime will request VM from VM cache server
```

Comment:
Change the introduction of vmcache in cli/config/configuration-qemu.toml.in to:
# Factory grpccache is the VMCache client.  It will request GRPC format
# VM and convert it back to a VM.  If VMCache function is enabled,
# kata-runtime will request VM from factory grpccache when it creates
# a new sandbox.